tests/foreman/api/test_product.py

```diff
@@ -35,22 +36,87 @@
 from robottelo.constants.repos import FAKE_1_PUPPET_REPO
 from robottelo.constants.repos import FAKE_1_YUM_REPO
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
 from robottelo.helpers import read_data_file
-from robottelo.test import APITestCase
 
 
-class ProductTestCase(APITestCase):
-    """Tests for ``katello/api/v2/products``."""
+def _org(request):
+    api = entities.Organization()
+    org = api.create()
+
+    @request.addfinalizer
+    def _cleanup():
+        # Delete organization if it still exists
+        if org.id in [o.id for o in api.search()]:
+            org.delete()
+
+    return org
+
+
+def _product(request, **kwargs):
+    api = entities.Product(**kwargs)
+    product = api.create()
+
+    @request.addfinalizer
+    def _cleanup():
+        # Delete product if it still exists
+        if product.id in [p.id for p in api.search()]:
+            product.delete()
+
+    return product
+
+
+def _gpg_key(request, **kwargs):
+    api = entities.GPGKey(**kwargs)
+    gpg_key = api.create()
+
+    @request.addfinalizer
+    def _cleanup():
+        # Delete GPG key if it still exists
+        if gpg_key.id in [g.id for g in api.search()]:
+            gpg_key.delete()
+
+    return gpg_key
+
+
+def _repository(request, **kwargs):
+    api = entities.Repository(**kwargs)
+    repo = api.create()
+
+    @request.addfinalizer
+    def _cleanup():
+        # Delete repo if it still exists
+        if repo.id in [r.id for r in api.search()]:
+            repo.delete()
+
+    return repo
+
+
+def _http_proxy(request, **kwargs):
+    api = entities.HTTPProxy(**kwargs)
+    http_proxy = api.create()
+
+    @request.addfinalizer
+    def _cleanup():
+        # Delete HTTP proxy if it still exists
+        if http_proxy.id in [h.id for h in api.search()]:
+            http_proxy.delete()
+
+    return http_proxy
 
-    @classmethod
-    def setUpClass(cls):
-        """Set up organization for tests."""
-        super().setUpClass()
-        cls.org = entities.Organization().create()
+
+@pytest.fixture(scope='module')
+def org(request):
```

Comment:
I see in a few of your PR's you have created such a structure for creating and deleting entities within fixtures.

There are some common entities already defined in `pytest_fixtures/api_fixtures.py`, which I think you should use here, keeping this conversion simpler.

As @swadeley has raised on this PR, there is a separate discussion to be had around which entities we should cleanup, and which should be left alone.

Reply:
Done.